vscode-extension/src/enhanced-extension.ts

```
@@ -0,0 +1,477 @@
+    } catch (error) {
+        vscode.window.showErrorMessage(`Failed to switch mode: ${error instanceof Error ? error.message : 'Unknown error'}`);
+        // Revert the settings
+        await config.update('useGateway', currentUseGateway, vscode.ConfigurationTarget.Global);
+        await config.update('preferCLI', currentPreferCLI, vscode.ConfigurationTarget.Global);
+        await enhancedMemoryService.refreshClient();
+    }
+}
+
+// Import existing API key management functions from original extension
+// These remain the same as they don't need CLI integration
+export async function manageApiKeys(apiKeyService: ApiKeyService) {
```

ESLint

[error] 'apiKeyService' is defined but never used. Allowed unused args must match /^_/u.

(@typescript-eslint/no-unused-vars)